indexing-service/src/main/java/org/apache/druid/indexing/overlord/setup/CategorizedWorkerBehaviorConfig.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.overlord.setup;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.indexing.overlord.autoscaling.AutoScaler;
+import org.apache.druid.indexing.worker.config.WorkerConfig;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This configuration allows overlord to work with several autoscalers to run tasks of different categories.
+ */
+public class CategorizedWorkerBehaviorConfig implements WorkerBehaviorConfig
+{
+  // Use the same category constant as for worker category to match default workers and autoscalers
+  public static final String DEFAULT_AUTOSCALER_CATEGORY = WorkerConfig.DEFAULT_CATEGORY;
+
+  private final WorkerSelectStrategy selectStrategy;
+  private final List<AutoScaler> autoScalers;
+
+  @JsonCreator
+  public CategorizedWorkerBehaviorConfig(
```

Comment:
I think things will be simplified if we removed this class and instead made  `DefaultWorkerBehaviorConfig` have `List<AutoScaler> autoScalers` with constructor

```
  @JsonCreator
  public DefaultWorkerBehaviorConfig(
      @JsonProperty("selectStrategy") WorkerSelectStrategy selectStrategy,
      @JsonProperty("autoScaler") AutoScaler autoScaler,
      @JsonProperty("autoScalers") List<AutoScaler> autoScalers
  )
  {
    this.selectStrategy = selectStrategy;
    // fail if both "autoscalers" and "autoscaler"  fields are non-null
    this.autoScalers = autoScalers or Collections.singletonList(autoscaler);
  }
```

Reply:
Do you thing it will not confuse users? I'm about having two sections related to autoscalers in the same config. 
I'm ok for that. Initially I also had such decision to just extend DefaultWorkerBehaviorConfig. But later I thought that having two sections for autoscalers can be not good for user experience.

Reply:
Actually, We are keeping `autoScaler` field only for backward compatibility .. we should really deprecate it and remove its mention from the docs. Eventually it should be removed and we should just have `autoScalers` .

Reply:
Ok. No objections. Let's move the `autoScalers` section to `DefaultWorkerBehaviorConfig` and remove `CategorizedWorkerBehaviorConfig`.

Reply:
I tried to refactor the code based on the above review comments. 
The class CategorizedWorkerBehaviorConfig has been removed, its logic been merged into the DefaultworkerBehaviorConfig. 
The constructor has been modified according to himanshug's post above.

One thing I noticed recently, is that the new web-console is not treating the behaviour config as a single json object but has separate text boxes for select strategy and autoscaler.

What shall we do about this?

<img width="604" alt="overlord-dynamic-config" src="https://user-images.githubusercontent.com/1635350/75812158-05aa2f80-5d8e-11ea-90a5-3a5c64947078.png">

Reply:
That means, users who want to use  `autoScalers` can't use the console UI to set the spec and would instead have to manually send the http request to coordinator using or curl or whatever.
after this PR is merged, an issue should be created for console UI updates to make it possible to add `autoScalers` .
In this PR, we need to make sure that we stay totally backwards compatible.

Reply:
got it. thanks.

Reply:
I am totally down to update the web console to reflect the new functionality.